src/modules/fancyzones/lib/util.cpp

```diff
@@ -209,6 +209,14 @@ void SaveWindowSizeAndOrigin(HWND window) noexcept
 
 void RestoreWindowSize(HWND window) noexcept
```

Comment:
function name is misleading now, perhaps rename to `RestoreWindowSizeIfNotMaximized`?

Reply:
I can do that, but in general, we have a lot of functions that have several checks at their beginning and if those checks don't pass we abort further processing. Renaming them to include those checks in the title, is certainly a bad practice. Maybe this one is specific because it is just one simple condition, but I would leave it as it is.

Reply:
Then we should do the check outside of the function, since having functions which silently do multiple things at once isn't good practice either.

Reply:
In `OnSnapHotkey` we call:
```
                        RestoreWindowOrigin(window);
                        RestoreWindowSize(window);
```
I suspect we should not call `RestoreWindowOrigin` in this case and check for `SW_SHOWMAXIMIZED` before calling those two functions.
Let's move the check outside of the function (there is another place where we call `RestoreWindowSize` that will have to do the same check).

Reply:
I think it is not possible for window to be maximized and we end up in that if statement, but no problem I'll add check both there and in `WindowMoveHandler`.

Reply:
If you can verify it, just add a comment, not code that we know it won't be executed.

Reply:
Well I'm not sure how to test it, but just simply looking at what that code does: handle win + arrow, and once we reached the last zone in single monitor environment, it doesn't cycle to the first one, but rather restores window to its origin and (pre-zoned) size. This means that window was in some zone previously, for sure not maximized.

Reply:
What if you start from a maximized window and you snap it to a zone with `Win + right arrow` and then unsnap it with `Win + left arrow`?

Reply:
That is an interesting case, but here is what will happen:
1. Maximize window.
2. Snap window to a zone with `win + right arrow`. We save his maximized size and origin.
3. Unsnap window from a zone using `win + left arrow`. Window will go back to its original size (maximized) and origin (top-left corner) but it will not have `maximized` capability so to say, because we only store window size and window origin, not other window placement information.

Reply:
We can consider that as an improvement in separate ticket. Store WINDOWPLACEMENT structure instead of just size and origin, because in we have everything:
```typedef struct tagWINDOWPLACEMENT {
  UINT  length;
  UINT  flags;
  UINT  showCmd;
  POINT ptMinPosition;
  POINT ptMaxPosition;
  RECT  rcNormalPosition;
  RECT  rcDevice;
} WINDOWPLACEMENT;
```